File: slider/src/main/java/com/adsamcik/slider/IntValueSlider.java
package com.adsamcik.slider;

import android.content.Context;
import android.content.SharedPreferences;
import android.support.annotation.NonNull;
import android.util.AttributeSet;

public class IntValueSlider extends ValueSlider<Integer> {
	public IntValueSlider(Context context) {
		super(context);
	}

	public IntValueSlider(Context context, AttributeSet attrs) {
		super(context, attrs);
	}

	public IntValueSlider(Context context, AttributeSet attrs, int defStyleAttr) {
		super(context, attrs, defStyleAttr);
	}

	public IntValueSlider(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
		super(context, attrs, defStyleAttr, defStyleRes);
	}

	@Override
	public void loadProgress(@NonNull SharedPreferences sharedPreferences, @NonNull String preferenceString, @NonNull Integer defaultValue) {
		super.setProgress(getItemIndex(sharedPreferences.getInt(preferenceString, defaultValue)));
	}

	@Override
	protected void updatePreferences(@NonNull SharedPreferences sharedPreferences, @NonNull String preferenceString, @NonNull Integer value) {
		sharedPreferences.edit().putInt(preferenceString, value).apply();
	}

}

File: slider/src/main/java/com/adsamcik/slider/sliders/IntValueSlider.kt
package com.adsamcik.slider.sliders

import android.content.Context
import android.content.SharedPreferences
import android.util.AttributeSet

class IntValueSlider : ValueSlider<Int> {
    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    override fun loadProgress(sharedPreferences: SharedPreferences, preferenceString: String, defaultValue: Int) {
        super.setProgress(getValueIndex(sharedPreferences.getInt(preferenceString, defaultValue)))
    }

    override fun updatePreferences(sharedPreferences: SharedPreferences, preferenceString: String, value: Int) {
        sharedPreferences.edit().putInt(preferenceString, value).apply()
    }

}
